### PR TITLE
Enable osx builds for ESMF and ESMPy

### PR DESCRIPTION
**Pull request by bekozi**

### CODE REVIEW

File: esmf/meta.yaml [hunk 1]
```diff
@@ -4,7 +4,6 @@ package:
 
 build: 
   number: 0
```
Comment (ocefpaf) [inline esmf/meta.yaml]: @bekozi you need to bump the build number to get a new Linux build.  Obvious-CI won't build packages that are already in the channel.

**Comment by bekozi** [comment]
@ocefpaf, @pelson: Do you have any suggestions for addressing the osx build issue on Travis CI? Prepending ``${PREFIX}/lib:${PREFIX}/lib64`` to ``DYLD_LIBRARY_PATH`` is one possible fix.

**Comment by pelson** [comment]
There is something very strange about the failing build:

```
The following packages will be downloaded:
    package                    |            build
    ---------------------------|-----------------
    krb5-1.13.2                |                0         1.2 MB  defaults
    openmpi-1.6.3              |                0         7.2 MB  defaults
    openssl-1.0.1k             |                1         2.5 MB  defaults
    curl-7.43.0                |                1         391 KB  defaults
    hdf5-1.8.15.1              |                2         1.6 MB  defaults
    libnetcdf-4.3.3.1          |                1         926 KB  defaults
    netcdf-fortran-4.4.2       |                0         610 KB  ioos
    ------------------------------------------------------------
                                           Total:        14.4 MB
The following NEW packages will be INSTALLED:
    curl:           7.43.0-1   defaults
    hdf5:           1.8.15.1-2 defaults
    krb5:           1.13.2-0   defaults
    libnetcdf:      4.3.3.1-1  defaults
    netcdf-fortran: 4.4.2-0    ioos    
    openmpi:        1.6.3-0    defaults
    openssl:        1.0.1k-1   defaults
    zlib:           1.2.8-0    defaults
Removing old work directory
Cloning into bare repository '/Users/travis/miniconda/conda-bld/git_cache/__git.code.sf.net_p_esmf_esmf'...
remote: Counting objects: 232848, done.
remote: Compressing objects: 100% (49828/49828), done.
remote: Total 232848 (delta 181276), reused 232818 (delta 181261)
Receiving objects: 100% (232848/232848), 189.06 MiB | 5.98 MiB/s, done.
Resolving deltas: 100% (181276/181276), done.
Checking connectivity... done.
checkout: u'ESMF_6_3_0rp1'
Cloning into '/Users/travis/miniconda/conda-bld/work'...
done.
Note: checking out 'ESMF_6_3_0rp1'.
You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.
If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:
  git checkout -b new_branch_name
HEAD is now at 5f824af... Add clang configurations to the all combinations table.
==> git log -n1 <==
commit 5f824aff569b904cce4ad854a14a4a71a35767d4
Author: Gerhard Theurich <theurich>
Date:   Thu Jul 10 22:27:17 2014 -0700
    Add clang configurations to the all combinations table.
==> git describe --tags --dirty <==
ESMF_6_3_0rp1
==> git status <==
HEAD detached at ESMF_6_3_0rp1
nothing to commit, working directory clean
Package: esmf-6.3.0rp1-1
source tree in: /Users/travis/miniconda/conda-bld/work
++ pwd
+ export ESMF_DIR=/Users/travis/miniconda/conda-bld/work
+ ESMF_DIR=/Users/travis/miniconda/conda-bld/work
+ export ESMF_INSTALL_PREFIX=/Users/travis/miniconda/envs/_build
+ ESMF_INSTALL_PREFIX=/Users/travis/miniconda/envs/_build
+ export ESMF_NETCDF=split
+ ESMF_NETCDF=split
+ export ESMF_NETCDF_INCLUDE=/Users/travis/miniconda/envs/_build/include
+ ESMF_NETCDF_INCLUDE=/Users/travis/miniconda/envs/_build/include
+ export ESMF_NETCDF_LIBPATH=/Users/travis/miniconda/envs/_build/lib
+ ESMF_NETCDF_LIBPATH=/Users/travis/miniconda/envs/_build/lib
++ uname
+ '[' Darwin == Darwin ']'
+ export ESMF_COMM=openmpi
+ ESMF_COMM=openmpi
+ make -j 2
Cannot open configuration file /opt/anaconda1anaconda2anaconda3/share/openmpi/mpicxx-wrapper-data.txt
Error parsing data file mpicxx: Not found
Cannot open configuration file /opt/anaconda1anaconda2anaconda3/share/openmpi/mpicxx-wrapper-data.txt
Error parsing data file mpicxx: Not found
Cannot open configuration file /opt/anaconda1anaconda2anaconda3/share/openmpi/mpif90-wrapper-data.txt
Error parsing data file mpif90: Not found
```

```/opt/anaconda1anaconda2anaconda3/``` is the prefix that conda-build uses as a placeholder prefix, and is switched for the real prefix at install time by conda. This clearly isn't happening for one (or more) of the packages being installed.

Some digging:

```
$ conda create -n delme2 -c ioos netcdf-fortran openmpi
    curl:           7.43.0-1   defaults
    hdf5:           1.8.15.1-2 defaults
    krb5:           1.13.2-0   defaults
    libnetcdf:      4.3.3.1-1  defaults
    netcdf-fortran: 4.4.2-0    ioos    
    openssl:        1.0.1k-1   defaults
    zlib:           1.2.8-0    defaults
...
$ cd /Users/pelson/miniconda/envs/delme2
$ grep anaconda1anaconda2 * -r
Binary file bin/otfconfig matches
Binary file bin/vtc++ matches
Binary file bin/vtcc matches
Binary file bin/vtcxx matches
Binary file bin/vtf77 matches
Binary file bin/vtf90 matches
Binary file bin/vtfort matches
Binary file bin/vtwrapper matches
Binary file lib/krb5/plugins/kdb/db2.so matches
Binary file lib/krb5/plugins/preauth/otp.so matches
Binary file lib/libcrypto.a matches
Binary file lib/libgssapi_krb5.2.2.dylib matches
Binary file lib/libgssapi_krb5.2.dylib matches
Binary file lib/libgssapi_krb5.dylib matches
Binary file lib/libkadm5clnt.dylib matches
Binary file lib/libkadm5clnt_mit.9.0.dylib matches
Binary file lib/libkadm5clnt_mit.9.dylib matches
Binary file lib/libkadm5clnt_mit.dylib matches
Binary file lib/libkadm5srv.dylib matches
Binary file lib/libkadm5srv_mit.9.0.dylib matches
Binary file lib/libkadm5srv_mit.9.dylib matches
Binary file lib/libkadm5srv_mit.dylib matches
Binary file lib/libkdb5.8.0.dylib matches
Binary file lib/libkdb5.8.dylib matches
Binary file lib/libkdb5.dylib matches
Binary file lib/libkrb5.3.3.dylib matches
Binary file lib/libkrb5.3.dylib matches
Binary file lib/libkrb5.dylib matches
Binary file lib/libmca_common_sm.3.dylib matches
Binary file lib/libmca_common_sm.dylib matches
Binary file lib/libmpi.1.dylib matches
Binary file lib/libmpi.dylib matches
Binary file lib/libmpi_cxx.1.dylib matches
Binary file lib/libmpi_cxx.dylib matches
Binary file lib/libmpi_f77.1.dylib matches
Binary file lib/libmpi_f77.dylib matches
Binary file lib/libompitrace.0.dylib matches
Binary file lib/libompitrace.dylib matches
Binary file lib/libopen-pal.4.dylib matches
Binary file lib/libopen-pal.dylib matches
Binary file lib/libopen-rte.4.dylib matches
Binary file lib/libopen-rte.dylib matches
Binary file lib/libotf.1.dylib matches
Binary file lib/libotf.dylib matches
Binary file lib/libotfaux.0.dylib matches
Binary file lib/libotfaux.dylib matches
Binary file lib/libvt-hyb.0.dylib matches
Binary file lib/libvt-hyb.a matches
Binary file lib/libvt-hyb.dylib matches
Binary file lib/libvt-mpi-unify.0.dylib matches
Binary file lib/libvt-mpi-unify.a matches
Binary file lib/libvt-mpi-unify.dylib matches
Binary file lib/libvt-mpi.0.dylib matches
Binary file lib/libvt-mpi.a matches
Binary file lib/libvt-mpi.dylib matches
Binary file lib/libvt-mt.0.dylib matches
Binary file lib/libvt-mt.a matches
Binary file lib/libvt-mt.dylib matches
Binary file lib/libvt.0.dylib matches
Binary file lib/libvt.a matches
Binary file lib/libvt.dylib matches
Binary file lib/openmpi/mca_io_romio.so matches
Binary file sbin/kadmind matches
Binary file sbin/kprop matches
Binary file sbin/kpropd matches
Binary file sbin/krb5kdc matches
```

Suggests that there are a number of files which aren't being built correctly.

I've dug even further to find that there are at least 2 packages which exhibit this bug:

```
$ ls -ltr */lib/libkadm5srv_mit.9.0.dylib
-rw-rw-r--  4 pelson  staff  168772 10 Jun 22:01 krb5-1.13.2-0/lib/libkadm5srv_mit.9.0.dylib
$ otool -L krb5-1.13.2-0/lib/libkadm5srv_mit.9.0.dylib
krb5-1.13.2-0/lib/libkadm5srv_mit.9.0.dylib:
	libkadm5srv_mit.9.0.dylib (compatibility version 9.0.0, current version 9.0.0)
	@loader_path/./libgssrpc.4.2.dylib (compatibility version 4.0.0, current version 4.2.0)
	@loader_path/./libgssapi_krb5.2.2.dylib (compatibility version 2.0.0, current version 2.2.0)
	@loader_path/./libkdb5.8.0.dylib (compatibility version 8.0.0, current version 8.0.0)
	@loader_path/./libkrb5.3.3.dylib (compatibility version 3.0.0, current version 3.3.0)
	@loader_path/./libk5crypto.3.1.dylib (compatibility version 3.0.0, current version 3.1.0)
	@loader_path/./libkrb5support.1.1.dylib (compatibility version 1.0.0, current version 1.1.0)
	@loader_path/./libcom_err.3.0.dylib (compatibility version 3.0.0, current version 3.0.0)
	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 46.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 159.1.0)
$ cat krb5-1.13.2-0/info/has_prefix
bin/compile_et
bin/krb5-config
lib/pkgconfig/gssrpc.pc
lib/pkgconfig/kadm-client.pc
lib/pkgconfig/kadm-server.pc
lib/pkgconfig/kdb.pc
lib/pkgconfig/krb5-gssapi.pc
lib/pkgconfig/krb5.pc
lib/pkgconfig/mit-krb5-gssapi.pc
lib/pkgconfig/mit-krb5.pc

$ otool -L ../../pkgs/openmpi-1.6.3-0/lib/libmpi.dylib 
../../pkgs/openmpi-1.6.3-0/lib/libmpi.dylib:
	/opt/anaconda1anaconda2anaconda3/lib/libmpi.1.dylib (compatibility version 2.0.0, current version 2.6.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 111.0.0)
	/usr/lib/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```

These are bugs at the Continuum end, and aren't the easiest thing to work around (short of building ourselves, or applying the same patches that conda-build applies).

I'll raise an issue on https://github.com/ContinuumIO/anaconda-issues.

**Comment by ocefpaf** [comment]
Phil, thanks for looking into this. I would never have figured this out on
my own.

Ben, thanks again for the recipes. Let's try to work this out with
Continuum.

**Comment by bekozi** [comment]
Sounds good guys. Thanks for looking into it!

**Comment by bekozi** [comment]
@ocefpaf: We thought it made sense to get serial OSX builds of ESMF and ESMPy in the channel. Should I open a ticket for OSX parallel builds and reference the outstanding issue with OpenMPI?

**Comment by ocefpaf** [comment]
> @ocefpaf: We thought it made sense to get serial OSX builds of ESMF and
ESMPy in the channel.

Yep. Makes sense.

Should I open a ticket for OSX parallel builds and reference the
outstanding issue with OpenMPI?

Please do!

Sorry for not looking into this before and not fixing the doc issues you
opened.  My laptop is broken and I am working from my phone (and stolen
time from my wife's laptop) until next week.

**Comment by bekozi** [comment]
> Sorry for not looking into this before and not fixing the doc issues you opened. 

No problem. It will be good to get it in there eventually.

> My laptop is broken and I am working from my phone (and stolen time from my wife's laptop) until next week.

Yikes. Hope your machine gets back soon! That's a lot of typing on a very small keyboard.

**Comment by ocefpaf** [comment]
Autocomplete is my new best friend ;-)

**Comment by ocefpaf** [comment]
Sorry @bekozi!  Somehow I missed your changes to this branch :flushed: 
Merging...

**Pull request merged**